src/NServiceBus.NHibernate/RepeatedFailuresOverTimeCircuitBreaker.cs

```diff
@@ -0,0 +1,68 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading;
+    using Logging;
+
+    class RepeatedFailuresOverTimeCircuitBreaker : IDisposable
+    {
+        public RepeatedFailuresOverTimeCircuitBreaker(string name, TimeSpan timeToWaitBeforeTriggering, Action<Exception> triggerAction)
+        {
+            this.name = name;
+            this.triggerAction = triggerAction;
+            this.timeToWaitBeforeTriggering = timeToWaitBeforeTriggering;
+
+            timer = new Timer(CircuitBreakerTriggered);
+        }
+
+        public void Success()
+        {
+            var oldValue = Interlocked.Exchange(ref failureCount, 0);
+
+            if (oldValue == 0)
+            {
+                return;
+            }
+
+            timer.Change(System.Threading.Timeout.Infinite, System.Threading.Timeout.Infinite);
```

Comment:
This is a bit hard to read and capture the intent. Would it be a good idea to have a properly named/intent capturing private method? This is subjective of course so feel free to ignore if you disagree.